flax/linen/attention.py

```diff
@@ -72,14 +67,9 @@ def dot_product_attention(query: Array,
       for details.
 
   Returns:
-    Output of shape `[batch..., length, num_heads, v_depth_per_head]`.
+    Output of shape `[batch..., num_heads, q_length, kv_length]`.
   """
-  assert key.ndim == query.ndim == value.ndim, 'q, k, v must have same rank.'
-  assert query.shape[:-3] == key.shape[:-3] == value.shape[:-3], (
-      'q, k, v batch dims must match.')
-  assert query.shape[-2] == key.shape[-2] == value.shape[-2], (
-      'q, k, v num_heads must match.')
-  assert key.shape[-3] == value.shape[-3], 'k, v lengths must match.'
+  assert key.ndim == query.ndim, 'q, k, v must have same rank.'
   assert query.shape[-1] == key.shape[-1], 'q, k depths must match.'
```

Comment:
Shouldn't you keep the asserts comparing `query` and `key` for the other indices?